### PR TITLE
chore: use `HomeDir` as prefix of `credentialDir`

### PR DESCRIPTION
**Pull request by nic-6443**

### CODE REVIEW

File: internal/persistence/credential.go [hunk 1]
```diff
@@ -37,33 +35,33 @@ func SaveCredential(credential *Credential) error {
...
 		if err != nil {
 			return fmt.Errorf("failed to create config directory in %s: %s", dir, err)
 		}
 	}
 
-	file, err := os.Create(credentialFileLocation)
+	file, err := os.Create(credentialDir)
 	if err != nil {
-		return fmt.Errorf("failed create file in %s for credential: %s", credentialFileLocation, err)
+		return fmt.Errorf("failed create file in %s for credential: %s", credentialDir, err)
```
Comment (starsz) [inline internal/persistence/credential.go:48]:
```suggestion
		return fmt.Errorf("failed to create file in %s for credential: %s", credentialDir, err)
```

File: internal/persistence/credential.go [hunk 2]
```diff
@@ -37,33 +35,33 @@ func SaveCredential(credential *Credential) error {
...
 	}
 
 	write, err := file.Write(data)
 	if err != nil {
-		return fmt.Errorf("failed write credential to %s, %s", credentialFileLocation, err)
+		return fmt.Errorf("failed write credential to %s, %s", credentialDir, err)
```
Comment (starsz) [inline internal/persistence/credential.go:53]:
```suggestion
		return fmt.Errorf("failed to write credential to %s, %s", credentialDir, err)
```

File: internal/persistence/credential.go [hunk 3]
```diff
@@ -37,33 +35,33 @@ func SaveCredential(credential *Credential) error {
...
 	}
 	if write != len(data) {
-		return fmt.Errorf("failed write credential to %s", credentialFileLocation)
+		return fmt.Errorf("failed write credential to %s", credentialDir)
```
Comment (starsz) [inline internal/persistence/credential.go:56]:
```suggestion
		return fmt.Errorf("failed to write credential to %s", credentialDir)
```

File: internal/persistence/types.go [hunk 4]
```diff
@@ -34,11 +34,14 @@ var (
...
 	HomeDir = filepath.Join(os.Getenv("HOME"), ".api7cloud")
 	// TLSDir is the directory to store TLS certificates.
-	TLSDir string
+	TLSDir        string
+	credentialDir string
 )
 
 // Init initializes the persistence context.
 func Init() error {
+	credentialDir = filepath.Join(HomeDir, "credentials")
```
Comment (starsz) [inline internal/persistence/types.go:43]: What's the difference with the old code?

Comment (tokers) [reply internal/persistence/types.go:43]: @starsz Just optimize the code base: one place to fetch the `HomeDir`. We just have two places to use `os.Getenv("HOME")` before, now we just have one.

**Review by tokers — APPROVED** [review]

**Review by jwrookie — APPROVED** [review]

File: internal/persistence/types.go (hunk 4 above)
Comment (nic-6443) [reply internal/persistence/types.go:43]: The variable `HomeDir` was not referenced before, but directly using `os.GetEnv()`. Now use `HomeDir` consistently as a prefix for other file directories.

Comment (starsz) [reply internal/persistence/types.go:43]: Got it.

**Review by starsz — APPROVED** [review]

**Pull request merged**